wagtail_localize_git/git.py

```diff
@@ -45,7 +45,25 @@ def pull(self):
             self.repo_is_empty = True
 
     def push(self):
-        self.gitpython.remotes.origin.push([f"refs/heads/{DEFAULT_BRANCH}"])
+        # GitPython push() returns a list of PushInfo instance for each head used in
+        # the push. The PushInfo instance will have any errors added to flags.
+        # In case of total failure, the returned list is empty.
+        refs = self.gitpython.remotes.origin.push([f"refs/heads/{DEFAULT_BRANCH}"])
+        if not refs:
+            return False
+        push_info = refs[0]
+        if any(
+            [
+                push_info.flags & push_info.ERROR,
+                push_info.flags & push_info.DELETED,
+                push_info.flags & push_info.REJECTED,
+                push_info.flags & push_info.REMOTE_FAILURE,
+                push_info.flags & push_info.REMOTE_REJECTED,
+            ]
+        ):
+            return False
+
+        return True
```

Comment:
Confession: I rarely use bitwise operators so I'm reading this code and its not totally obvious to me if this looks like it works or not :smile: 

We're testing a layer out by mocking `_push` to throw/not throw `SyncPushError`s at various times but this logic isn't really directly under test. It would be reassuring if we could add an additional test case or two to the `TestRepositoryClonePullPush` class which exercises this logic directly and checks the return values.


Also, as a style point: We can write code of the form

```py
if some_condition:
    return False
return True
```

as

```py
return not some_condition
```

but I'll leave it with you to write it as you think is most readable :)

Reply:
Thanks @chris48s, I agree on the need for the extra tests. Will add 'em.

I do try to follow that style point where possible. The earlier iteration of this was simpler, but at the same time did not cover all the possible errors.

Will amend when adding the logic specific tests 👍